### PR TITLE
XML removal (part 13)

### PR DESCRIPTION
**Pull request by fgalan**
Part of #1862 

Exceptionally, I'd recommend to review this PR commit-by-commit (at the "diff" tab doesn't show all the changes).

### CODE REVIEW

File: archive/xml/README.md [hunk 1]
```diff
@@ -5,7 +5,8 @@ In particular, the xmlCheck checker (which was used to check that XML in unittes
 and test harness fragments were compliant with NGSI XSD).
 
 In addition, the following XML-based unit tests were removed. If you want to have a look to them,
-have a look to code just before merging PR #X (this is the commit number: X).
+have a look to code just before merging PR #2104 (this is the commit number:
```
Comment (kzangeli) [inline archive/xml/README.md]: have a look to => have a look at

File: test/unittests/convenience/Convenience_test.cpp [hunk 2]
```diff
@@ -70,97 +70,3 @@ TEST(Convenience, emptyPath)
...
-
-
-
-/* ****************************************************************************
-*
-* badPathNgsi10 - 
-*
-* FIXME P5 #1862: _json countepart?
-*/
-TEST(Convenience, DISABLED_badPathNgsi10)
```
Comment (kzangeli) [inline test/unittests/convenience/Convenience_test.cpp]: These three tests seem simple to move to json.
However, as they're disabled, perhaps something is going on ...

Comment (fgalan) [inline test/unittests/convenience/Convenience_test.cpp]: Ok... I'll try to rescue these 3 from oblivion :)

File: archive/xml/README.md (hunk 1 above)
Comment (fgalan) [inline archive/xml/README.md]: Fixed in 70e119f

File: test/unittests/convenience/Convenience_test.cpp (hunk 2 above)
Comment (fgalan) [inline test/unittests/convenience/Convenience_test.cpp]: Fixed in 70e119f

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**